weak_supervision/data/dataset_readers/semantic_parsing/latent_alignment.py

```diff
@@ -8,34 +8,51 @@
 from allennlp.data.dataset_readers.dataset_reader import DatasetReader
 from allennlp.data.fields import TextField, Field, ListField, MetadataField
 from allennlp.data.instance import Instance
-from allennlp.data.tokenizers import Token, WordTokenizer
+from allennlp.data.tokenizers import Token, Tokenizer, WordTokenizer
 from allennlp.data.tokenizers.word_splitter import JustSpacesWordSplitter
-from allennlp.data.token_indexers import SingleIdTokenIndexer
+from allennlp.data.token_indexers import SingleIdTokenIndexer, TokenIndexer
 
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
+def preprocess_tokens(token): 
+    if token.startswith('fb:row.row.'):
+        return token.replace('fb:row.row.', '')
+    elif token.startswith('fb:cell.'):
+        return token.replace('fb:cell.', '')
+    elif token.startswith('fb:part.'):
+        return token.replace('fb:part.', '')
+    else:
+        return token
+
+
 @DatasetReader.register("latent_alignment")
 class LatentAlignmentDatasetReader(DatasetReader):
     def __init__(self,
-                 dpd_directory: str,
+                 tokenizer: Tokenizer = None,
+                 utterance_token_indexers: Dict[str, TokenIndexer] = None,
+                 logical_form_token_indexers: Dict[str, TokenIndexer] = None,
                  max_logical_forms: int = 500,
+                 process_tokens: bool = False, 
                  lazy: bool = False) -> None:
         super().__init__(lazy)
-        self._dpd_directory = dpd_directory
         self._max_logical_forms = max_logical_forms
-        self._utterance_token_indexers = {'tokens': SingleIdTokenIndexer(namespace='tokens')}
-        self._logical_form_token_indexers = {'lf_tokens': SingleIdTokenIndexer(namespace='lf_tokens')}
-        self._tokenizer = WordTokenizer(JustSpacesWordSplitter())
+        self._utterance_token_indexers = utterance_token_indexers or {'tokens': SingleIdTokenIndexer(namespace='tokens')}
+        self._logical_form_token_indexers = logical_form_token_indexers or {'lf_tokens': SingleIdTokenIndexer(namespace='lf_tokens')}
+
+        self._process_tokens = process_tokens
+        self._tokenizer = tokenizer or WordTokenizer(JustSpacesWordSplitter())
 
     @overrides
     def _read(self, file_path: str):
         with open(cached_path(file_path), "r") as data_file:
             examples = json.load(data_file)
         for utterance, sempre_forms in examples:
             if len(sempre_forms) > self._max_logical_forms:
-                sempre_forms.sort(key=len)
-                sempre_forms = sempre_forms[:self._max_logical_forms]
+                sempre_form_gold = sempre_forms[0]
```

Comment:
You need to deal with the first logical form specially only for dev data, because that is when you have the gold logical form at the top, right?

Reply:
Yes! Is there a way to check if this is dev or train? I just decided to follow the same logic because it doesn't affect training